src/models/routineDao.js

```diff
@@ -98,14 +100,42 @@ const getRoutineHistoryByDate = async (userId, startDate, endDate) => {
       routines.created_at < ?
     GROUP BY routines.id
     ;
-  `, [userId, startDate, endDate]);
+  `,
+    [userId, startDate, endDate]
+  );
   return routine;
-}
+};
+
+const checkExercisesIdInRoutine = async (routineId, exercisesId) => {
+  const result = await AppDataSource.query(
+    `SELECT * 
+    FROM routine_exercises
+    WHERE routine_id = ? AND exercise_id IN (?)`,
+    [routineId, exercisesId]
+  );
+
+  return result;
+};
+
+// 아래 내용  -  회의 후 변경 예정 : 무조건 상태완료로 변경하는 게 아닌, 완료 체크 후 보내준 것에 대해서만 상태완료로 변경하는 쪽으로
+//  →  기존에 완료한 운동도 체크 해제하면 미완료로 변경 가능하게
+const updateCompletedExerciseStatusbyRoutineId = async (
+  routineId,
+  exercisesId
+) => {
+  await AppDataSource.query(
+    `UPDATE routine_exercises
+    SET completed = 1
+    WHERE routine_id = ? AND exercise_id IN (?)`,
+    [routineId, exercisesId]
+  );
+};
```

Comment:
```suggestion
//  →  기존에 완료한 운동도 체크 해제하면 미완료로 변경 가능하게

const exerciseStatus = {
  COMPLETED : 1
  NOT_COMPLETED : 0
}
const updateCompletedExerciseStatusbyRoutineId = async (
  routineId,
  exercisesId
) => {
  await AppDataSource.query(
    `UPDATE routine_exercises
    SET completed = exerciseStatus.COMPLETED
    WHERE routine_id = ? AND exercise_id IN (?)`,
    [routineId, exercisesId]
  );
};
```

1. 그냥 completed = 1 이라고 하면, 1과 0의 상태가 어떠한 경우인지 모든 개발자가 외우고 있거나, 참조가 필요하게 됩니다. 숫자를 값으로 매핑하면, 커뮤니케이션 시간을 줄일 수 있고, human error가 발생할 확률을 줄일 수 있습니다 :)

2. 더 나아가, 보다 확장성 있는 함수를 만들 수 있는 방법은, status (상태) 에 대한 값도 함수의 인자로 받아오게 함수를 선언하고, status를 service layer에서 내려주는 방법이 있습니다.

Reply:
매 요청마다 모든 상태가 변경되도록 수정했습니다.
routine_id에 속하는 exercise_id를 받으면 해당 운동만 1(운동 완료), 나머지는 0(운동 미완료)로 변경됩니다.
 + 말씀해주신 부분도 반영 완료

변경 함수

const updateCompletedExerciseStatusbyRoutineId = async (id, exercisesId) => {
  const exerciseStatus = {
    COMPLETED: 1,
    NOT_COMPLETED: 0,
  };

  await AppDataSource.query(
    `UPDATE routine_exercises
      SET completed = IF(exercise_id IN (?), ${exerciseStatus.COMPLETED}, ${exerciseStatus.NOT_COMPLETED})
    WHERE routine_id = ?`,
    [exercisesId, id]
  );
};